application/api/answer/routes.py

```diff
@@ -168,53 +169,66 @@ def complete_stream(question, retriever, conversation_id):
-    if 'prompt_id' in data:
-        prompt_id = data["prompt_id"]
-    else:
-        prompt_id = 'default'
-    if 'selectedDocs' in data and data['selectedDocs'] is None:
-        chunks = 0
-    elif 'chunks' in data:
-        chunks = int(data["chunks"])
-    else:
-        chunks = 2
-    
-    prompt = get_prompt(prompt_id)
+    try:
+        data = request.get_json()
+        # get parameter from url question
+        question = data["question"]
+        if "history" not in data:
+            history = []
+        else:
+            history = data["history"]
+            history = json.loads(history)
+        if "conversation_id" not in data:
+            conversation_id = None
+        else:
+            conversation_id = data["conversation_id"]
+        if 'prompt_id' in data:
+            prompt_id = data["prompt_id"]
+        else:
+            prompt_id = 'default'
+        if 'selectedDocs' in data and data['selectedDocs'] is None:
+            chunks = 0
+        elif 'chunks' in data:
+            chunks = int(data["chunks"])
+        else:
+            chunks = 2
 
-    # check if active_docs is set
+        prompt = get_prompt(prompt_id)
 
-    if "api_key" in data:
-        data_key = get_data_from_api_key(data["api_key"])
-        source = {"active_docs": data_key["source"]}
-    elif "active_docs" in data:
-        source = {"active_docs": data["active_docs"]}
-    else:
-        source = {}
 
-    if source["active_docs"].split("/")[0] == "default" or source["active_docs"].split("/")[0] == "local":
-        retriever_name = "classic"
-    else:
-        retriever_name = source['active_docs']
 
-    retriever = RetrieverCreator.create_retriever(retriever_name, question=question, 
-        source=source, chat_history=history, prompt=prompt, chunks=chunks, gpt_model=gpt_model
-        )
+        # check if active_docs is set
+
+        if "api_key" in data:
+            data_key = get_data_from_api_key(data["api_key"])
+            source = {"active_docs": data_key["source"]}
+        elif "active_docs" in data:
+            source = {"active_docs": data["active_docs"]}
+        else:
+            source = {}
 
-    return Response(
-        complete_stream(question=question, retriever=retriever,
-                        conversation_id=conversation_id), mimetype="text/event-stream")
+        if source.get("active_docs", "").split("/")[0] in ["default", "local"]:
+            retriever_name = "classic"
+        else:
+            retriever_name = source["active_docs"]
+
+        retriever = RetrieverCreator.create_retriever(retriever_name, question=question, 
+                                                      source=source, chat_history=history, 
+                                                      prompt=prompt, chunks=chunks, 
+                                                      gpt_model=gpt_model)
+
+        return Response(
+            stream_with_context(complete_stream(question, retriever,
```

Comment:
hi @sarfarazsiddiquii 
the  `chat_history` argument is being passed here in the `complete_stream` function without specifying it in the function definition which is throwing an error

Reply:
Thanks @siiddhantt for addressing the error, it seems like `chat_history` and `prompt_id `was unnecessary regarding this issue.
I may have overlooked the above `create_retriever` function while making changes. I removed the unnecessary properties from `complete_stream` function and the recent commit should work without any errors.
Tell me if any further changes are required.

Reply:
No worries, I'd recommend running the code and testing the flow as I encountered this error at the very start. Do let me know when you've tested it.

Reply:
hey @siiddhantt , I have made some minor changes since my last commits, and after running the code as requested, I did not encounter any errors, and the code worked fine. Please review the commits, and if you encounter any errors, please let me know what errors you're getting. 
Thank you.